### ISSUE TITLE
Some question about tutorial spleen_segmentation_3d_lightning

### ISSUE THREAD

**Issue by cvlearn913** [Project-MONAI/tutorials#514]
**Describe the bug**
I encountered the following problem when I tried to run the tutorial code ["spleen_segmentation_3d_lightning"](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb)
![image](https://user-images.githubusercontent.com/59678550/150275661-bb77c01f-2008-44e5-b14a-3b956596860a.png)
The code can run successfully in a single card, but it get failed when I change it to distributed

**To Reproduce**
Here is my change 
before
```python
trainer = pytorch_lightning.Trainer(
    gpus=[0],
    max_epochs=600,
    logger=tb_logger,
    enable_checkpointing=True,
    num_sanity_val_steps=1,
    log_every_n_steps=16,
)
```

After 
```python
    trainer = pytorch_lightning.Trainer(
        gpus=[0,1],
        max_epochs=600,
        strategy='dp',
        logger=tb_logger,
        checkpoint_callback=True,
        num_sanity_val_steps=1,
        check_val_every_n_epoch=100
    
    )
```

**Environment (please complete the following information):**
 - OS: linux  5.13.0-25-generic
 - Python version：3.7
 - MONAI version MONAI rev id: a7835ab5cdc3859c0f06c66ab4888a16b51ea712
 - CUDA/cuDNN version 11.1 
 - GPU models and configuration Double RTX 3090(24GB）

**Additional context**
When I debug it, I found that the Tensor is on a different GPU，
![image](https://user-images.githubusercontent.com/59678550/150276767-c321d11b-53e5-4ebd-8578-ab4f9e58fb8c.png)
so it raised an error here in monai/metrics/metric.py line 256.
![image](https://user-images.githubusercontent.com/59678550/150276814-9e778817-9da0-4335-85a0-aa2f04681152.png)
How should I use distributed training in lightning framework with MONAI?

**Comment by Nic-Ma** [comment]
Hi @cvlearn913 ,

Thanks for your interest here.
I am not very familiar with Lightning framework, @marksgraham could you please help share some help here?

Thanks in advance.

**Comment by marksgraham** [comment]
Hi,

I must admit I've moved away from Lightning and haven't used it for a long time, so am much less familiar with it now. But according to the advice [here](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html) can you try to add a `type_as` to the forward pass:

```
def forward(self, x):
     return self._model(x).type_as(x)
```

**Comment by cvlearn913** [comment]
Thanks for your reply, 
```python
def forward(self, x):
     return self._model(x).type_as(x)
```
but same problem after modification
![image](https://user-images.githubusercontent.com/59678550/150908022-2e06fbee-4ef8-43f9-bbbb-3c25223d2cfc.png)

**Comment by cvlearn913** [comment]
![image](https://user-images.githubusercontent.com/59678550/151319090-9bebf4a6-3cfc-476d-ab37-2e0b47c0b0dc.png)
Thanks for your replies ！ @Nic-Ma ，@marksgraham  I modified part of the code to avoid this problem temporarily,
```python
    def __init__(self):
        super().__init__()
        self._model = UNet(
            spatial_dims=3,
            in_channels=1,
            out_channels=classes,
            channels=(16, 32, 64, 128, 256),
            strides=(2, 2, 2, 2),
            num_res_units=2,
            norm=Norm.BATCH,
        )
        self.loss_function = DiceFocalLoss(to_onehot_y=True,sigmoid=True)
        self.post_pred = Compose([EnsureType(), Activations(sigmoid=True),AsDiscrete(threshold=0.5)])
        self.post_label = Compose([EnsureType(), Activations(sigmoid=True),AsDiscrete(threshold=0.5)])
        ####Separate mertric's calculations
        self.dice_metric_0 = DiceMetric(include_background=False, reduction="mean_batch", get_not_nans=False)
        self.dice_metric_1 = DiceMetric(include_background=False, reduction="mean_batch", get_not_nans=False)
        self.best_val_dice = 0
        self.best_val_epoch = 0
```

```python
    def validation_step(self, batch, batch_idx):
        images, labels = batch["image"], batch["label"]

        roi_size = (128, 128, 128)
        sw_batch_size = 4
        outputs = sliding_window_inference(
            images, roi_size, sw_batch_size, self.forward)


        loss = self.loss_function(outputs, labels)
        outputs = [self.post_pred(i) for i in decollate_batch(outputs)]
        labels = [self.post_label(i) for i in decollate_batch(labels)]
        if images.device.index == 0:###Dice coefficient is calculated explicitly
            self.dice_metric_0(y_pred=outputs, y=labels)

        else:
            self.dice_metric_1(y_pred=outputs, y=labels)

        return {"val_loss": loss, "val_number": len(outputs)}

    def validation_epoch_end(self, outputs):
        val_loss, num_items = 0, 0
        for output in outputs:
            val_loss += output["val_loss"].sum().item()
            num_items += output["val_number"]
        if outputs[0]["val_loss"].device.index ==0:
                dice_mertric = self.dice_metric_0
        else:
            dice_mertric = self.dice_metric_1
        mean_val_dice = dice_mertric.aggregate().item()
        dice_mertric.reset()
        mean_val_loss = torch.tensor(val_loss / num_items)
        tensorboard_logs = {
            "val_dice": mean_val_dice,
            "val_loss": mean_val_loss,
        }
        if mean_val_dice > self.best_val_dice:
            self.best_val_dice = mean_val_dice
            self.best_val_epoch = self.current_epoch
        print(
            f"current epoch: {self.current_epoch} "
            f"current mean dice: {mean_val_dice:.4f}"
            f"\nbest mean dice: {self.best_val_dice:.4f} "
            f"at epoch: {self.best_val_epoch}"
        )
        return {"log": tensorboard_logs}
```
Now that distributed training is available, but what's wrong with calculating the DICE coefficient this way?
![image](https://user-images.githubusercontent.com/59678550/151319960-fe31d464-7725-4276-a9a4-51624d6d6b82.png)

**Comment by BelieferQAQ** [comment]
def validation_step(self, batch, batch_idx):
        images, labels = batch["image"], batch["label"]

        roi_size = (128, 128, 128)
        sw_batch_size = 4
        outputs = sliding_window_inference(
            images, roi_size, sw_batch_size, self.forward)


        **loss = self.loss_function(outputs, labels)**

您好，请问你的验证集损失可以计算出来吗，outputs和labels的shape是相同的吧。

**Comment by KumoLiu** [comment]
Hi @BelieferQAQ，
The shape of input and output after `sliding_window_inference` is the same, so the shape of outputs and labels is also identical.  You can also use loss metric to compute validation loss, like loss in https://github.com/pytorch/ignite/blob/master/ignite/metrics/loss.py

**Comment by BelieferQAQ** [comment]
![image](https://user-images.githubusercontent.com/91660759/179476835-d8a13ed3-572d-4158-8948-6839dfc37b08.png)

**Comment by BelieferQAQ** [comment]
i try monai version 0.6，0.7， 0.9，but this is do not work
